src/apps/persistence/tools.clj

```diff
@@ -32,10 +33,11 @@
                      :tool_type_id
                      :interactive]))
 
-(defn- get-tool-type-id
-  "Gets the ID of the given tool type name."
-  [tool-type]
-  (:id (first (select tool_types (fields :id) (where {:name tool-type})))))
+(defn- get-tool-type-id-for-tool
+  "Gets the tool type ID to use for a tool."
+  [{:keys [interactive type]}]
+  (when-let [tool-type (if interactive c/interactive-tool-type type)]
+    (:id (first (select tool_types (fields :id) (where {:name tool-type}))))))
```

Comment:
This change was intended to ensure that the interactive flag and the tool type are consistent.